src/Api/InitializeInTracee.cpp

```diff
@@ -4,16 +4,47 @@
 
 #include "Api/InitializeInTracee.h"
 
+#include <functional>
+
 #include "Api/Orbit.h"
-#include "OrbitBase/Result.h"
+#include "OrbitBase/ExecutablePath.h"
+#include "OrbitBase/UniqueResource.h"
+#include "UserSpaceInstrumentation/Attach.h"
+#include "UserSpaceInstrumentation/ExecuteInProcess.h"
+#include "UserSpaceInstrumentation/InjectLibraryInTracee.h"
 
+using namespace orbit_user_space_instrumentation;
+using orbit_grpc_protos::ApiTableInfo;
 using orbit_grpc_protos::CaptureOptions;
 
 namespace orbit_api {
+ErrorMessageOr<void> InitializeInTracee(const CaptureOptions& capture_options) {
+  if (capture_options.api_table_infos().size() == 0) {
+    return ErrorMessage("No api table to initilize.");
+  }
+
+  ErrorMessageOr<void> result = outcome::success();
+  int32_t pid = capture_options.pid();
+
+  OUTCOME_TRY(AttachAndStopProcess(pid));
+  {
+    // Make sure we resume the target process, even on early-outs.
+    orbit_base::unique_resource scope_exit{
+        pid, [&result](int32_t pid) { result = DetachAndContinueProcess(pid); }};
+
+    // Load liborbit.so and find api table initialization function.
+    const std::string kLibPath = orbit_base::GetExecutableDir() / "liborbit.so";
+    const char* kInitFunction = "orbit_api_initialize";
+    OUTCOME_TRY(handle, DlopenInTracee(pid, kLibPath, RTLD_NOW));
+    OUTCOME_TRY(address, DlsymInTracee(pid, handle, kInitFunction));
+
+    // Initialize all api function tables.
+    for (const ApiTableInfo& info : capture_options.api_table_infos()) {
```

Comment:
This loop could be moved into the init function, right? Not sure if it is worth it.

Reply:
Oh but we'd have to pass in arrays to the init function, seems complicated, or am I missing something?

Reply:
Hmm, yes, right. You would need to allocate memory in the tracee for that and pass the pointer. Let's keep it as is. If we really need this at some point we'll come up with some tooling around it.